maths/prime_numbers.py

```diff
@@ -13,16 +13,16 @@ def primes(max: int) -> List[int]:
     >>> primes(1_000_000)[-1]
     999983
     """
-    max += 1
-    numbers = [False] * max
-    ret = []
-    for i in range(2, max):
-        if not numbers[i]:
-            for j in range(i, max, i):
-                numbers[j] = True
-            ret.append(i)
-    return ret
+    numbers: Generator = (i for i in range(1, (max + 1)))
+    for i in filter(lambda x: x > 1, numbers):
+        for j in range(2, i):
+            if (i % j) == 0:
+                break
+        else:
+            yield i
 
 
 if __name__ == "__main__":
-    print(primes(int(input("Calculate primes up to:\n>> "))))
+    number = int(input("Calculate primes up to:\n>> "))
```

Comment:
As discussed in CONTRIBUTING.md, use __str.strip()__ to gracefully deal with leading or trailing whitespace in user input.  __number = int(input("Calculate primes up to:\n>> ").strip())__

Reply:
fixed